homeassistant/components/dlna_dmr/media_player.py

```diff
@@ -608,11 +634,80 @@ def media_position_updated_at(self) -> datetime | None:
         return self._device.media_position_updated_at
 
     @property
-    def unique_id(self) -> str:
-        """Report the UDN (Unique Device Name) as this entity's unique ID."""
-        return self.udn
+    def media_artist(self) -> str | None:
+        """Artist of current playing media, music track only."""
+        if not self._device:
+            return None
+        return self._device.media_artist
 
     @property
-    def usn(self) -> str:
-        """Get the USN based on the UDN (Unique Device Name) and device type."""
-        return f"{self.udn}::{self.device_type}"
+    def media_album_name(self) -> str | None:
+        """Album name of current playing media, music track only."""
+        if not self._device:
+            return None
+        return self._device.media_album_name
+
+    @property
+    def media_album_artist(self) -> str | None:
+        """Album artist of current playing media, music track only."""
+        if not self._device:
+            return None
+        return self._device.media_album_artist
+
+    @property
+    def media_track(self) -> int | None:
+        """Track number of current playing media, music track only."""
+        if not self._device:
+            return None
+        return self._device.media_track_number
+
+    @property
+    def media_series_title(self) -> str | None:
+        """Title of series of current playing media, TV show only."""
+        if not self._device:
+            return None
+        return self._device.media_series_title
+
+    @property
+    def media_season(self) -> str | None:
+        """Season number, starting at 1, of current playing media, TV show only."""
+        if not self._device:
+            return None
+        # Some DMRs, like Kodi, leave this as 0 and encode the season & episode
+        # in the episode_number metadata, as {season:d}{episode:02d}
+        if (
+            not self._device.media_season_number
+            or self._device.media_season_number == "0"
+        ) and self._device.media_episode_number:
+            try:
+                episode = int(self._device.media_episode_number, 10)
+                if episode > 100:
+                    return str(episode // 100)
+            except ValueError:
+                pass
+        return self._device.media_season_number
+
+    @property
+    def media_episode(self) -> str | None:
+        """Episode number of current playing media, TV show only."""
+        if not self._device:
+            return None
+        # Complement to media_season math above
+        if (
+            not self._device.media_season_number
+            or self._device.media_season_number == "0"
+        ) and self._device.media_episode_number:
+            try:
+                episode = int(self._device.media_episode_number, 10)
+                if episode > 100:
+                    return str(episode % 100)
+            except ValueError:
+                pass
```

Comment:
```suggestion
            with contextlib.suppress(ValueError):
                episode = int(self._device.media_episode_number, 10)
                if episode > 100:
                    return str(episode % 100)
```

In a future PR you could switch this to contextlib.suppress

Reply:
Good tip, I'd forgotten about `contextlib.suppress`, and can use it in a few other places too.